### ISSUE TITLE
error message when attempting to enable/disable ice mode mode on lg fridge

### ISSUE THREAD

**Issue by olivebadger14** [ollo69/ha-smartthinq-sensors#589]
**Describe the bug**
error message when attempting to enable/disable ice plus mode mode on lg fridge

**Expected behavior**
when i toggle ice plus on/off i expect it to turn on/off accordingly with no errors

**Screenshots**
![IMG_2735](https://github.com/ollo69/ha-smartthinq-sensors/assets/110454154/3f863204-5b5b-4451-8630-c884c7cacf2c)

**Environment details:**
 - Environment:  dedicated mini pc with Home Assistant OS
 - Home Assistant version installed: 2023.08.01
 - Component version installed: 0.33.1
 - Last know working version: cant exactly remember but it was before v 0.31.7
 - LG device type and model with issue: LG fridge GTF916PZPYD
 - LG devices connected (list): fridge only

**Output of HA logs**
Paste the relavant output of the HA log here.

```
he above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 226, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1974, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2011, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 235, in handle_service
    return await service.entity_service_call(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 870, in entity_service_call
    response_data = await _handle_entity_call(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 942, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/smartthinq_sensors/switch.py", line 298, in async_turn_off
    self._api.async_set_updated()
  File "/config/custom_components/smartthinq_sensors/__init__.py", line 453, in async_set_updated
    self._coordinator.async_set_updated_data(self._state)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 401, in async_set_updated_data
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 164, in async_update_listeners
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 431, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 746, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 808, in _async_write_ha_state
    state = self._stringify_state(available)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 752, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 595, in state
    raise ValueError(
ValueError: Sensor sensor.refrigerator_freezer_temp has device class 'temperature', state class 'measurement' unit '°C' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: '' (<class 'str'>)
2023-08-24 08:51:42.156 ERROR (SyncWorker_8) [homeassistant.components.influxdb] Cannot connect to InfluxDB due to 'HTTPConnectionPool(host='192.168.1.7', port=8086): Max retries exceeded with url: /write?db=grafana_db (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f50e0a76010>: Failed to establish a new connection: [Errno 111] Connection refused'))'. Please check that the provided connection details (host, port, etc.) are correct and that your InfluxDB server is running and accessible. Retrying in 60 seconds.
2023-08-24 08:52:01.029 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 591, in state
    numerical_value = int(value)
                      ^^^^^^^^^^
ValueError: invalid literal for int() with base 10: ''

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 229, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 374, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 164, in async_update_listeners
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 431, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 746, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 808, in _async_write_ha_state
    state = self._stringify_state(available)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 752, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 595, in state
    raise ValueError(
ValueError: Sensor sensor.refrigerator_freezer_temp has device class 'temperature', state class 'measurement' unit '°C' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: '' (<class 'str'>)
2023-08-24 08:52:32.424 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 591, in state
    numerical_value = int(value)
                      ^^^^^^^^^^
ValueError: invalid literal for int() with base 10: ''

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 229, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 374, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 164, in async_update_listeners
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 431, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 746, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 808, in _async_write_ha_state
    state = self._stringify_state(available)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 752, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 595, in state
    raise ValueError(
ValueError: Sensor sensor.refrigerator_freezer_temp has device class 'temperature', state class 'measurement' unit '°C' and suggested precision 'None' thus indicating it has a numeric value; however, it has the non-numeric value: '' (<class 'str'>)

```

**Comment by ollo69** [comment]
@olivebadger14,

there is an issue template that should be filled with all requested information and is always better to attach integration diagnostic.
Please provide all required information or this issue will be ignored and than closed.

Thanks👍

**Comment by olivebadger14** [comment]
no problem, i ll edit the bug report

**Comment by ollo69** [comment]
Can you please attach device diagnostic?

**Comment by olivebadger14** [comment]
[smartthinq_sensors-afa57ea0b1e32258f014e36cb45637c7-REFRIGERATOR-2ebb6e51ea67b33fed29b46cb8db86a2.json.txt](https://github.com/ollo69/ha-smartthinq-sensors/files/12447907/smartthinq_sensors-afa57ea0b1e32258f014e36cb45637c7-REFRIGERATOR-2ebb6e51ea67b33fed29b46cb8db86a2.json.txt)

**Comment by ollo69** [comment]
Please try if last release fix the issue.

**Comment by olivebadger14** [comment]
hello i can confirm that the issue is resolved with v 0.33.3

**Comment by ollo69** [comment]
Good, I close this issue.
